### ISSUE TITLE
ToggleButton widget not updating props "active" on test

### ISSUE THREAD

**Issue by timvandingenen1993** [KlausSchaefers/quant-ux#158]
Hi,

the ToggleButton widget doesn't seem to update it's props & style in it's qModel info when clicked in test mode. 
It does when you do it in the edit mode.

any workaround or fix for this issue?

**Comment by KlausSchaefers** [comment]
Hi,

just to be clear. In a script you want to check if an active button is active or not? So something like

```

let screen = qux.getScreen('Screen')
let button = screen.getWidget('ToggleButton')

if (button.props.active === true) {
....
}

```

The state of the segment widget and the toggle bound can be bound to a variable. if you bind the button to "a", a script like this should be able to change the state

```
data.a = !data.a
```

**Comment by timvandingenen1993** [comment]
Yes exactly I was trying:
`button.qModel.props.active === true `

I understand your approach with data binding, I just found it strange that it updates in the props when you change them in edit mode, I would expect it to also work in the way I tried.

**Comment by KlausSchaefers** [comment]
Yes, that would make sense. For all widget the main state is stored in a variable called 'value', so should be

```
button.setProp({'value', true})
```

Mutating the qModel will not have any effect, you have to go through the setProp methods, so the APOI can track the changes.

I should freeze the model maybe.
